test/core/nexus/NexusRemoteTests.cpp

```diff
@@ -25,77 +25,78 @@ class Nexus_Remote_Test : public ::testing::Test {
+{
+    MPI_Init(NULL, NULL);
+    MPI_Comm_rank(MPI_COMM_WORLD, &mpi_rank);
 
-   downstream_remote_nexus = std::make_shared<HY_PointHydroNexusRemoteDownstream>(nexus_id_number, nexus_id, num_downstream);
+    stored_discharge.push_back(1.2);
+    stored_discharge.push_back(1.8);
+    stored_discharge.push_back(2.3);
+}
 
+void Nexus_Remote_Test::TearDown()
+{
+    MPI_Finalize();
 }
 
-void Nexus_Remote_Test::TearDown() {
 
-}
 
 //Test sending data with MPI from an upstream remote nexus
 //to a downstream remote nexus.
 TEST_F(Nexus_Remote_Test, TestInit0)
 {
-   int send_catchment_id = 27;
-   double send_flow = 1.2;
-   long send_time_step = 3;
-   
-   int receive_catchment_id = 0;
-   double receive_flow = 0.0;
-   long receive_time_step = 0;
-
-   // Find out rank, size
-   int world_rank;
-   MPI_Comm_rank(MPI_COMM_WORLD, &world_rank);
-
-   // If the world_rank is 0, then call HY_PointHydroNexusRemoteUpstream to send 
-   // data to downstream remote nexus.
-   if (world_rank == 0) 
-   {
-      upstream_remote_nexus->add_upstream_flow(send_flow, send_catchment_id, send_time_step);
-   }
+    HY_PointHydroNexusRemote::catcment_location_map_t loc_map;
 
-   // Else if the world_rank is 1, then call HY_PointHydroNexusRemoteDownstream to receive data
-   // from the upstream remote nexus.
-   else if (world_rank == 1) 
-   {
-      downstream_remote_nexus->add_upstream_flow(NULL, NULL, NULL);
+    std::shared_ptr<HY_PointHydroNexusRemote> nexus;
 
-      receive_catchment_id = downstream_remote_nexus->get_catchment_id();
-      receive_flow = downstream_remote_nexus->get_upstream_flow_value();
-      receive_time_step = downstream_remote_nexus->get_time_step();
-
-      std::cout << "Received: \nCatchment ID " << receive_catchment_id << " \nFlow " 
-      << receive_flow << " \nTime Step " << receive_time_step << std::endl;
-
-      //Assert equal for the send and receive values 
-      EXPECT_DOUBLE_EQ(send_flow, receive_flow);
+    // create a nexus at both ranks
+    if ( mpi_rank == 0)
+    {
+        loc_map[26] = 1;
+        nexus = std::make_shared<HY_PointHydroNexusRemote>(27, "nexus-27", 1, loc_map);
+    }
+    else if ( mpi_rank == 1)
+    {
+        loc_map[27] == 0;
+        nexus = std::make_shared<HY_PointHydroNexusRemote>(26, "nexus-26", 1, loc_map);
+    }
 
-      EXPECT_EQ(send_catchment_id, receive_catchment_id);
+    double dummy_flow = -9999.0;
+    long ts = 0;
 
-      EXPECT_EQ(send_time_step, receive_time_step);
+    for ( auto discharge : stored_discharge)
+    {
+        switch(mpi_rank)
+        {
+            case 0:
+                std::cerr << "Rank 0: Sending flow of " << discharge << " to catchment 26\n";
+                nexus->add_upstream_flow(discharge,1,ts);
+                nexus->get_downstream_flow(26,ts,100);
+            break;
+
+            case 1:
+                nexus->add_upstream_flow(dummy_flow,27,ts);
+                double recieved_flow = nexus->get_downstream_flow(2,ts,100);
```

Comment:
shouldn't `get_downstream_flow(...) `  be  26 or  27?  2 doesn't make too much sense...

Reply:
The nexus code was not designed for nexus to nexus communication. It will look odd when we misuse it like this.